modules/data.land/R/InventoryGrowthFusion.R

```diff
@@ -294,22 +393,42 @@ model{
                       tau_dbh = 1, 
                       tau_inc = 1500,
                       tau_ind = 50, 
-                      tau_yr = 100, 
+                      tau_yr = 100,
+                      betaX2 = 0, 
                       ind = rep(0, data$ni),  
                       year = rep(0, data$nt))
   }
   
-  ## compile JAGS model
+
+  PEcAn.utils::logger.info("COMPILE JAGS MODEL")
   j.model <- jags.model(file = textConnection(TreeDataFusionMV), data = data, inits = init, n.chains = 3)
-  ## burn-in
-  jags.out <- coda.samples(model = j.model, 
-                           variable.names = burnin.variables, 
-                           n.iter = min(n.iter, 2000))
+
+  PEcAn.utils::logger.info("BURN IN")
+ # jags.out <- coda.samples(model = j.model, 
+ #                          variable.names = burnin.variables, 
+ #                          n.iter = min(n.iter, 2000))
   if (burnin_plot) {
     plot(jags.out)
   }
   
-  ## run MCMC
-  coda.samples(model = j.model, variable.names = out.variables, n.iter = n.iter)
+  PEcAn.utils::logger.info("RUN MCMC")
+  load.module("dic")
+  for(k in seq_len(ceiling(n.iter/n.chunk))){
+    if(k%%50 == 0){
+	vnames <- c("x",out.variables)   ## save x periodically
+    } else {
+	vnames <- out.variables
+    }
+    jags.out <- coda.samples(model = j.model, variable.names = vnames, n.iter = n.chunk)
+    ofile <- paste("IGF",model,k,"RData",sep=".")
+    print(ofile)
+    save(jags.out,file=ofile)
+    ## could add code here to check for convergence and break from loop early
+    D <- as.mcmc.list(lapply(jags.out,function(x){x[,'deviance']}))
+    gbr <- coda::gelman.diag(D)$psrf[1,1]
+    trend <- mean(sapply(D,function(x){coef(lm(x~seq_len(n.chunk)))[2]}))
+    if(gbr < 1.005 & abs(trend) < 0.5) break
+  }
```

Comment:
1.005 seems like a really strict convergence criterion. Isn't it more common to use something around 1.1?

Also, don't you want to do all of this on your cumulative samples? It seems like this could lead to a lot of extra iterations if there's high autocorrelation in the chains relative to `n.chunk`.

Also, it might be nice to print `k / ceiling(n.iter/n.chunk)` occasionally, since it's a nice progress indicator.